webrecorder/browsertrix
# Edit crawl config as YAML

(https://github.com/webrecorder/browsertrix-cloud/issues/203) Support editing crawl config as YAML.

### Manual testing
1. Run app and go to new crawl template page
2. Toggle "Advanced Editor". Verify code editor is shown
3. Type and paste in config as plain JSON or YAML. Verify code editor works as expected
4. Enter in invalid YAML. Verify error is shown and you're prevented from creating the crawl template
5. Fix invalid YAML. Verify crawl template is created with expected crawl config
6. Repeat test for editing config

### Screenshots
**YAML:**
<img width="640" alt="Screen Shot 2022-04-05 at 2 56 33 PM" src="https://user-images.githubusercontent.com/4672952/161856463-5d281411-defc-413a-9cc9-77c5d728b4e6.png">

**YAML error handling:**
<img width="629" alt="Screen Shot 2022-04-05 at 2 56 54 PM" src="https://user-images.githubusercontent.com/4672952/161856485-48610db6-e9fd-49c7-bd79-200cc2aab242.png">

**Crawl template detail view:**
<img width="640" alt="Screen Shot 2022-04-05 at 3 05 37 PM" src="https://user-images.githubusercontent.com/4672952/161857644-6142f542-5eb1-4d96-940d-55b45dc2b2eb.png">


### Ideas
The yaml parse errors are pretty nice. Could implement friendly JSON parse errors.
